### PR TITLE
Respect sorting from the file list in the preview app

### PR DESCRIPTION
**Pull request by JammingBen**
## Description

* Adds sorting to the preview app
* Refactors the code and splits up `App.vue` into smaller components

## Related Issue

- Fixes https://github.com/owncloud/web/issues/7978
- Fixes https://github.com/owncloud/web/issues/3821

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>acceptance oCIS</strong> https://drone.owncloud.com/owncloud/web/34691/66/1
:boom: The oCISSharingPublic2 tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>acceptance oCIS</strong> https://drone.owncloud.com/owncloud/web/34691/64/1
:boom: The oCISSharingPerm2 tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>acceptance oC10</strong> https://drone.owncloud.com/owncloud/web/34691/71/1
:boom: The oC10IntegrationApp2 tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>acceptance oC10</strong> https://drone.owncloud.com/owncloud/web/34691/70/1
:boom: The oC10IntegrationApp1 tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>acceptance oCIS</strong> https://drone.owncloud.com/owncloud/web/34691/68/1
:boom: The oCISUploadMove tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>acceptance oCIS</strong> https://drone.owncloud.com/owncloud/web/34691/65/1
:boom: The oCISSharingPublic1 tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>acceptance oCIS</strong> https://drone.owncloud.com/owncloud/web/34691/69/1
:boom: The oCISTrashbinJourney tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>acceptance oCIS</strong> https://drone.owncloud.com/owncloud/web/34691/67/1
:boom: The oCISSharingPublic3 tests pipeline failed. The build has been cancelled.

File: packages/web-app-preview/src/App.vue [hunk 1]
```diff
@@ -233,15 +161,47 @@ export default defineComponent({
...
+        unref(unref(currentFileContext).space),
+        unref(activeFilteredFile)
+      )
+      router.replace({
+        ...unref(route),
+        params: { ...unref(route).params, ...params },
+        query: { ...unref(route).query, ...query }
+      })
+    }
+
```
Comment (dschmidt) [inline packages/web-app-preview/src/App.vue:193]: I really like where this PR is going! 

It doesn't need to be implemented in this PR and it probably makes sense to do it, when we introduce another app that needs it, but long term I would really like to separate this logic from the app.

Comment (JammingBen) [reply packages/web-app-preview/src/App.vue:193]: I'm not 100% sure where your comment belongs to, it kinda sits at the end of `updateLocalHistory` - you mean that method in general? 😄

**Review by kulmann — APPROVED** [review]
Woah nice 😍

File: packages/web-app-preview/src/App.vue (hunk 1 above)
Comment (dschmidt) [reply packages/web-app-preview/src/App.vue:193]: Nah, all of that file switching, route, filtering logic. 

I would like to implement just the filter in this app and have eVerything else handled by a composable

**Pull request merged**